### PR TITLE
Create posts

### PR DESCRIPTION
**Pull request by lucianocaus**
Screenshot 2024-01-05 at 14 14 17
Screenshot 2024-01-05 at 14 14 24

### CODE REVIEW

File: src/components/Button/index.js [hunk 1]
```diff
@@ -25,6 +27,7 @@ Button.propTypes = {
   buttonClassName: PropTypes.string,
   text: PropTypes.string.isRequired,
   disabled: PropTypes.bool,
+  onClick: PropTypes.func,
```
Comment (Allison1795) [inline src/components/Button/index.js:30]: it should be required, right? or add a default value

File: src/components/Input/index.js [hunk 2]
```diff
@@ -18,4 +19,10 @@ const Input = forwardRef(({
   />
 ));
 
+Input.propTypes = {
+  placeholder: propTypes.string,
+  id: propTypes.string.isRequired,
+  type: propTypes.string,
```
Comment (Allison1795) [inline src/components/Input/index.js:25]: if placeholder and type are not required, you should add default values

File: src/components/NewPost/index.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import { useNavigate } from 'react-router-dom';
+import { useForm } from 'react-hook-form';
+import { toast } from 'react-toastify';
+
+import { createPost } from 'api/post.service';
+
+import Input from 'components/Input';
+import Button from 'components/Button';
+import Box from '../Box';
+import TextArea from '../TextArea';
```
Comment (Allison1795) [inline src/components/NewPost/index.js]:
```suggestion
import Box from 'components/Box';
import TextArea from 'components/TextArea';
```

File: src/components/NewPost/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+const NewPost = () => {
+  const { register, handleSubmit } = useForm();
+  const navigate = useNavigate();
+  const onSubmitHandler = ({
```
Comment (Allison1795) [inline src/components/NewPost/index.js]:
```suggestion
  const navigate = useNavigate();
  
  const onSubmitHandler = ({
```

File: src/components/Sidebar/index.js [hunk 5]
```diff
@@ -1,12 +1,19 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Modal from 'react-modal';
 
+import Button from '../Button';
 import Box from '../Box';
+import NewPost from '../NewPost';
```
Comment (Allison1795) [inline src/components/Sidebar/index.js]: use aliases here

File: src/components/TextArea/index.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    id={id}
+    type={type}
+    rows={rows}
+    cols={cols}
+    {...props}
+  />
+));
+
+TextArea.propTypes = {
+  placeholder: propTypes.string,
```
Comment (Allison1795) [inline src/components/TextArea/index.js:27]: if is not required, you should add a placeholder

File: src/components/Sidebar/index.js [hunk 7]
```diff
@@ -1,12 +1,19 @@
...
+import React, { useState } from 'react';
+import Modal from 'react-modal';
 
-import Box from '../Box';
+import Button from 'components/Button';
+import Box from 'components/Box';
+import NewPost from 'components/NewPost';
 
-import './style.scss';
+import style from './style.scss';
```
Comment (LucasBruzzone) [inline src/components/Sidebar/index.js:8]: I believe that if you do it this way, the style is not just added automatically to this page. So 'sidebar' or 'sidebar__actions' classnames' css won't be here

File: src/components/TextArea/index.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  id: propTypes.string.isRequired,
+  type: propTypes.string,
+  rows: propTypes.oneOfType([
+    propTypes.string,
+    propTypes.number,
+  ]),
+  cols: propTypes.oneOfType([
+    propTypes.string,
+    propTypes.number,
+  ]),
```
Comment (LucasBruzzone) [inline src/components/TextArea/index.js]: Why can it be both? Shouldn't it be just one?

**Review by LucasBruzzone — APPROVED** [review]

File: src/components/Sidebar/index.js (hunk 7 above)
Comment (lucianocaus) [reply src/components/Sidebar/index.js:8]: I think the styles are getting loaded, the issue with not using the `style` is that I need it in order to send it to the `Modal`

File: src/components/TextArea/index.js (hunk 8 above)
Comment (lucianocaus) [reply src/components/TextArea/index.js]: image

One of them was unnecessary

**Pull request merged**